realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -551,7 +582,9 @@ public void onChange() {
         RealmChangeListener listenerB = new RealmChangeListener() {
             @Override
             public void onChange() {
-                counter.incrementAndGet();
+                assertEquals(0, counter.get());
```

Comment:
Now we rely on the order of listener invocations. Worth to create another test case to ensure that ListenerA is called first.